tokio/src/runtime/task/mod.rs

```diff
@@ -214,6 +214,7 @@ unsafe impl<S> linked_list::Link for Task<S> {
     }
 
     unsafe fn pointers(target: NonNull<Header>) -> NonNull<linked_list::Pointers<Header>> {
-        NonNull::from(&mut *target.as_ref().owned.get())
+        // Not super great as it avoids some of looms checking...
+        NonNull::from(target.as_ref().owned.with_mut(|ptr| &mut *ptr))
```

Comment:
I'm thinking about tweaking the linked list a bit so that loom's `UnsafeCell` can check the pointers type, at some point.

Reply:
Would be good as a follow up PR once this lands.

Reply:
yup!